RecoLocalTracker/SiPixelRecHits/plugins/SiPixelRecHitHeterogeneous.cc

```diff
@@ -100,10 +111,10 @@ void SiPixelRecHitHeterogeneous::produceCPU(edm::HeterogeneousEvent& iEvent, con
   edm::Handle<SiPixelClusterCollectionNew> hclusters;
   iEvent.getByToken(clusterToken_, hclusters);
 
-  auto output = std::make_unique<SiPixelRecHitCollectionNew>();
-  run(hclusters, *output);
+  auto output = std::make_unique<CPUProduct>();
+  run(hclusters, output->collection);
 
-  output->shrink_to_fit();
+  output->collection.shrink_to_fit();
   iEvent.put(std::move(output));
```

Comment:
This should be changed to `iEvent.put<Output>(std::move(output))` to produce the `HeterogeneousProduct` for CPU.